### PR TITLE
ci: fetch data plan on release build

### PR DESCRIPTION
**Pull request by alexs-mparticle**
# Summary

- Adding an action that uses a [Data Plan Fetch Reusable Workflow](https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/data-plan-fetch.yml) will fetch the current data plan and opens a draft pull request against `development`.
-  This will be a "push button" action for the time being as Data Plans will not update frequently enough to warrant frequent polling.

# Master Issue

Closes https://go.mparticle.com/work/SQDSDKS-3683

### CODE REVIEW

**Review by mchuangatmp — CHANGES_REQUESTED** [review]
let's add the GPG step when committing so it's verified.

Also i am leaning toward we not put this work in the release yaml and into a separate daily yaml file

**Comment by alexs-mparticle** [comment]
I agree with the GPG, but I'm not sure about the Daily Check. Data plans don't upgrade frequently and since we're not validating, it feels excessive to check on a daily basis.

File: .github/workflows/sync-higgs-shop-data-plan.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        - name: Download Data Plan Artifacts
+          uses: actions/download-artifact@v3
+          with:
+            name: higgs-shop-dataplan
+            path: core-sdk-samples/higgs-shop-sample-app/dataplans
+
+        - name: Create Commit if Data Plan Changed
+          uses: stefanzweifel/git-auto-commit-action@v4
+          with:
+            commit_message: Update Higgs Shop Data Plan
```
Comment (mchuangatmp) [inline .github/workflows/sync-higgs-shop-data-plan.yml:41]: i'd add our semantic version prefix here

Comment (alexs-mparticle) [reply .github/workflows/sync-higgs-shop-data-plan.yml:41]: Good catch. Should it be `chore`?

File: .github/workflows/sync-higgs-shop-data-plan.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+name: Sync Higgs Shop Data Plan
```
[the remark this answers is not in the document]
Comment (alexs-mparticle) [reply .github/workflows/sync-higgs-shop-data-plan.yml:1]: Unclear at this point. It's something we should probably discuss once we have more than one sample app.

**Review by mchuangatmp — APPROVED** [review]
remember to squash and merge

**Pull request merged**